mapeditor/PlayerSelectionDialog.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * PlayerSelectionDialog.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#include "PlayerSelectionDialog.h"
+#include "mainwindow.h"
+#include "../lib/mapping/CMap.h"
+
+PlayerSelectionDialog::PlayerSelectionDialog(QWidget * parent)
+	: QDialog(parent), selectedPlayer(PlayerColor::NEUTRAL)
+{
+	auto main = qobject_cast<MainWindow *>(parent);
+	assert(main);
+
+	setupDialogComponents();
+
+	int maxPlayers = 0;
+	if(main && main->controller.map())
+		maxPlayers = main->controller.map()->players.size();
+	
+	bool defaultIsChecked = false;
+
+	for(int i = 0; i < maxPlayers; ++i)
+	{
+		PlayerColor player(i);
+		QAction * action = main->getActionPlayer(player);
+
+		addCheckbox(action, player, main->controller.map()->players.at(i).canAnyonePlay(), & defaultIsChecked);
+	}
+}
+
+void PlayerSelectionDialog::onCheckboxToggled(bool checked)
+{
+	if(!checked)
+		return;
+
+	QCheckBox * senderCheckBox = qobject_cast<QCheckBox *>(sender());
+	if(!senderCheckBox)
+		return;
+
+	for(int i = 0; i < checkboxes.size(); ++i)
+	{
+		QCheckBox * cb = checkboxes[i];
+		if(cb == senderCheckBox)
+		{
+			selectedPlayer = PlayerColor(i);
+		}
+		else
+		{
+			cb->blockSignals(true);
+			cb->setChecked(false);
+			cb->blockSignals(false);
```

Comment:
this logic sounds like you want an exclusive QButtonGroup. And maybe radiobuttons will suit better here?